pkg/clean/clean.go

```diff
@@ -27,39 +28,44 @@ var log = logger.Log
 // including security policy, static route, subnet, subnet port, subnet set, vpc, ip pool, nsx service account
 // it is usually used when nsx-operator is uninstalled and remove all the resources created by nsx-operator
 // return error if any, return nil if no error
-func Clean(cf *config.NSXOperatorConfig) (Status, error) {
+func Clean(ctx context.Context, cf *config.NSXOperatorConfig) (Status, error) {
 	log.Info("starting NSX cleanup")
-	if err := cf.ValidateConfigFromCmd(); err != nil {
-		return ValidationFailed, err
-	}
-	nsxClient := nsx.GetClient(cf)
-	if nsxClient == nil {
-		return GetNSXClientFailed, fmt.Errorf("failed to get nsx client")
-	}
-	if cleanupService, err := InitializeCleanupService(cf); err != nil {
-		return InitCleanupServiceFailed, fmt.Errorf("failed to initialize cleanup service: %w", err)
-	} else if cleanupService.err != nil {
-		return InitCleanupServiceFailed, cleanupService.err
-	} else {
-		for _, clean := range cleanupService.cleans {
-			if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
-				if err != nil {
-					log.Info("retrying to clean up NSX resources", "error", err)
-					return true
-				}
-				return false
-			}, func() error {
-				if err := clean.Cleanup(); err != nil {
-					return err
+	select {
```

Comment:
This isn't really doing anything, because it will only check on invocation of this function. We want the logic actually passed down the stack. Essentially, `clean.Cleanup()` should accept a `ctx` and further down the line.

Reply:
Actually, you want to support timeout in context, it can satisfy your need https://github.com/vmware-tanzu/nsx-operator/pull/489/files#diff-be6b68a2cf4b036867ce2ddc12ed2a9c862555110e0265a06b7d42ced61ec345R77, we think there is no need to accept a `ctx` in `clean.Cleanup()`, `clean.Cleanup() ` is our inner logic, you don't need to concern it.

Reply:
ok